--- Image/Adapter/Gd2.php	2025-10-06 08:06:36.000000000 +0000
+++ Image/Adapter/Gd2.php	2025-12-02 10:07:11.661327539 +0000
@@ -77,10 +77,21 @@
             throw new \OverflowException('Memory limit has been reached.');
         }
         $this->imageDestroy();
-        $this->_imageHandler = call_user_func(
-            $this->_getCallback('create', null, sprintf('Unsupported image format. File: %s', $this->_fileName)),
-            $this->_fileName
-        );
+        try{
+            $this->_imageHandler = call_user_func(
+                $this->_getCallback('create', null, sprintf('Unsupported image format. File: %s', $this->_fileName)),
+                $this->_fileName
+            );
+        }catch(\Exception $e){
+            if(strpos($e->getMessage(), 'imagecreatefrompng(): gd-png: libpng warning: iCCP: known incorrect sRGB profile') === false){
+                throw $e;
+            }
+
+            $this->_imageHandler = @call_user_func(
+                $this->_getCallback('create', null, sprintf('Unsupported image format. File: %s', $this->_fileName)),
+                $this->_fileName
+            );
+        }
     }
 
     /**
